src/components/Firework/Firework.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {FC, useEffect, useRef, useState} from 'react'
+import {Dimensions, View, Animated, Easing} from 'react-native'
+import {getRandomNumber} from './utils'
+
+import styles from './styles'
+
+type FireworkProps = {
+  density?: number
+  iterations?: number
+  duration?: number
+}
+type CoordsType = {
+  x: number[]
+  y: number[]
+}
+
+export const Firework: FC<FireworkProps> = ({
+  density = 5,
+  iterations = 30,
+  duration = 700,
+}) => {
+  const [coords, setCoords] = useState<CoordsType>({
+    x: [],
+    y: [],
+  })
+  const opacityAnim = useRef(new Animated.Value(1)).current
+  const moveAnim = useRef(new Animated.Value(0)).current
+
+  useEffect(() => {
+    setExplosionSpots()
+  }, [])
+
+  const animateOpacity = () => {
+    opacityAnim.setValue(1)
+    Animated.timing(opacityAnim, {
+      toValue: 0,
+      duration,
+      easing: Easing.ease,
+      useNativeDriver: false,
```

Comment:
you can try to preserve native driver usage (anmations would be smoother) if you use translationX/translationY instead of top/left/right/bottom (just a suggestion)

Reply:
Yeah, for using native driver I needed to replace `top/left` with `transform` values.
Good idea, thanks, because I received a warning or even an error that I could not use native driver and reason is I'm using improper properties.